src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessService.java

```diff
@@ -45,11 +43,7 @@ public CaseAccessService(@Qualifier(CachedUserRepository.QUALIFIER) UserReposito
     }
 
     public Boolean canUserAccess(CaseDetails caseDetails) {
-
-        IDAMProperties currentUser = userRepository.getUserDetails();
```

Comment:
This is an expensive call to IDAM that returns user id and roles. User id and roles are already available in the spring security context and has been implemented in the UserRepository.

Reply:
But @gbenadikar given that the UserRepository injected is a CachedUserRepository, which is caching the getUserDetails() method, aren't we retrieving that from the cache already?

Reply:
CaseViewType is request scope, doesn't survive beyond the request.

Reply:
yes correct but same is the spring security context right which we are switching to now

Reply:
spring security context holds the authentication token passed with the request and derives the user roles from it without making explicit call to IDAM. User.getUserDetails makes explicit IDAM API call to retrieve user roles and I found it to be expensive.